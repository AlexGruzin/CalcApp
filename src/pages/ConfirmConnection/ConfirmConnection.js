import React from 'react';
import PropTypes from 'prop-types';
import { OAUTH_ACCESS_STATUS } from 'constants/connections';
import Spinner from 'components/Spinner';
import { useTranslation } from 'react-i18next';

import './ConfirmConnection.styles.less';

const ConfirmConnection = ({ integrationStatus }) => {
  const { t } = useTranslation();

  return (
    <div className="confirm-connection">
      {!integrationStatus && (
        <>
          <Spinner />
          <div className="confirm-connection__text">{t('confirm:connection.processing')}</div>
        </>
      )}
      {integrationStatus === OAUTH_ACCESS_STATUS.SUCCESS && (
        <div className="confirm-connection__done">{t('confirm:success')}</div>
      )}
      {integrationStatus === OAUTH_ACCESS_STATUS.ERROR && (
        <div className="confirm-connection__block">{t('confirm:error')}</div>
      )}
    </div>
  );
};

ConfirmConnection.propTypes = {
  integrationStatus: PropTypes.string,
};

ConfirmConnection.defaultProps = {
  integrationStatus: '',
};

export default ConfirmConnection;
