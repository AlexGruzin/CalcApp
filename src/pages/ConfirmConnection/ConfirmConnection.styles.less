.confirm-connection {
  display: flex;
  flex-grow: 1;
  background: var(--white);
  justify-content: center;
  align-items: center;
}
.confirm-connection__status {
  position: absolute;
}

.confirm-connection__text {
  margin-top: var(--space-x10);
  color: var(--font-black);
}

.confirm-connection .spinner > div {
  border-color: var(--black);
  border-right-color: transparent;
}

.confirm-connection__block {
  color: var(--error-color);
}

.confirm-connection__done {
  color: var(--turquoise);
}
