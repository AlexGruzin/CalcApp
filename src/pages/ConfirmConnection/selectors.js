import { createStructuredSelector } from 'reselect';
import { getIntegrationStatus } from 'domains/connections/selectors';

export default createStructuredSelector({
  integrationStatus: getIntegrationStatus,
});
