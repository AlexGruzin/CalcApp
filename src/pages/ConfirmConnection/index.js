import { connect } from 'react-redux';
import ConfirmConnection from './ConfirmConnection';
import selector from './selectors';

export default connect(selector, null)(ConfirmConnection);
