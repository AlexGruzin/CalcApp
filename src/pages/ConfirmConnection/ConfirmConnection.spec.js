import React from 'react';
import { render } from '@testing-library/react';
import ConfirmConnection from './ConfirmConnection';

describe('ConfirmConnection tests', () => {
  it('Check snapshots of ConfirmConnection are equal.', () => {
    const { asFragment } = render(<ConfirmConnection />);
    expect(asFragment()).toMatchSnapshot();
  });
});
