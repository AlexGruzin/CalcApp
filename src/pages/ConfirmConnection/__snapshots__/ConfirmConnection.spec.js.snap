// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ConfirmConnection tests Check snapshots of ConfirmConnection are equal. 1`] = `
<DocumentFragment>
  <div
    class="confirm-connection"
  >
    <div
      class="spinner"
    >
      <div
        class="circle__Wrapper-sc-16bbsoy-0 dpSNSD"
        color="#fff"
        size="24"
      />
    </div>
    <div
      class="confirm-connection__text"
    >
      confirm:connection.processing
    </div>
  </div>
</DocumentFragment>
`;
